neo4j/work/summary.py

```diff
@@ -69,13 +66,12 @@ class ResultSummary:
 
     def __init__(self, **metadata):
         self.metadata = metadata
-        self.protocol_version = metadata.get("protocol_version")
         self.server = metadata.get("server")
         self.query = metadata.get("query")
         self.parameters = metadata.get("parameters")
         self.query_type = metadata.get("type")
         self.counters = SummaryCounters(metadata.get("stats", {}))
-        if self.protocol_version[0] < BOLT_VERSION_3:
+        if self.server.protocol_version[0] < BOLT_VERSION_3:
```

Comment:
Get with default values instead to avoid check on version?

Reply:
Added a comment TODO: This logic should be inside the Bolt subclasses, because it can change depending on Bolt Protocol Version.